src/Processors/Transforms/CreatingSetsTransform.cpp

```diff
@@ -118,14 +121,20 @@ void CreatingSetsTransform::startSubquery()
                     done_with_set = true;
                     set_from_cache = true;
                 }
-                catch (const Exception & e)
-                {
-                    /// Exception that is thrown by the future::get() is shared across all waiters and cannot be modified from multiple threads.
-                    /// Re-create exception to allow later multiple modify (i.e. addMessage() during pipeline execution)
-                    throw Exception(e);
-                }
+                break;
+            }
+            /// Exception that is thrown by the shared_future::get() is shared across all waiters and cannot be modified from multiple threads.
+            /// Re-create exception to allow later concurrent modify (i.e. addMessage() during pipeline execution)
+            ///
+            /// Note, that findOrPromiseToBuild() can also call shared_future::get()
+            catch (const Exception & e)
+            {
+                throw Exception(e);
+            }
+            catch (...)
+            {
+                throw Exception::createDeprecated(getExceptionMessage(std::current_exception(), /* with_stacktrace= */ false), ErrorCodes::UNKNOWN_EXCEPTION);
```

Comment:
isn't `createRuntime` better here?

Reply:
Haven't seen it. Though it does the same thing internally and comment states that it is for "third-party and network exceptions". But I don't have any objections

Reply:
yes it does, but I don't like what we are using explicitly "deprecated" function 😄

Reply:
I found it odd as well, but since it had been introduced with it's users and is used now as well in the code...
Anyway I've rebased the patch to use `createRuntime`.